test/images/logs-generator/logs_generator.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"strconv"
+	"time"
+)
+
+const (
+	linesTotalEnvVariableName = "LOGS_GENERATOR_LINES_TOTAL"
+	durationEnvVariableName   = "LOGS_GENERATOR_DURATION"
+)
+
+var (
+	httpMethods = []string{
+		"GET",
+		"POST",
+		"PUT",
+		"DELETE",
+	}
+	namespaces = []string{
+		"kube-system",
+		"default",
+		"my-custom-namespace",
+	}
+	resources = []string{
+		"pods",
+		"services",
+		"endpoints",
+		"configmaps",
+	}
+)
+
+func main() {
+	linesTotal, durationSeconds, err := parseGenerateParameters()
+	if err != nil {
+		log.Fatalf("Failed to extract parameters for generator from environment: %v", err)
+		os.Exit(1)
+	}
+
+	generateLogs(linesTotal, durationSeconds)
+}
+
+// Extracts parameters for logs generation from environment variables
+func parseGenerateParameters() (linesTotal int, durationSeconds int, err error) {
+	linesTotalStr, ok := os.LookupEnv(linesTotalEnvVariableName)
+	if !ok {
+		err = fmt.Errorf("Missing environment variable %s", linesTotalEnvVariableName)
+		return
+	}
+	if linesTotal, err = strconv.Atoi(linesTotalStr); err != nil {
+		err = fmt.Errorf("Error parsing %s: %v", linesTotalEnvVariableName, err)
+		return
+	}
+
+	durationStr, ok := os.LookupEnv(durationEnvVariableName)
+	if !ok {
+		err = fmt.Errorf("Missing environment variable %s", durationEnvVariableName)
+		return
+	}
+	duration, err := time.ParseDuration(durationStr)
+	if err != nil {
+		err = fmt.Errorf("Error parsing %s: %v", durationEnvVariableName, err)
+		return
+	}
+
+	durationSeconds = int(duration.Seconds())
+	if durationSeconds <= 0 {
+		err = fmt.Errorf("Invalid duration: %v, must be a positive number of seconds", duration)
+		return
+	}
+
+	return
+}
+
+// Outputs linesTotal lines of logs to stdout uniformly throughout durationSeconds seconds
+func generateLogs(linesTotal int, durationSeconds int) {
+	delay := time.Duration(float64(durationSeconds) / float64(linesTotal) * float64(time.Second))
+	randomSource := rand.NewSource(time.Now().UnixNano())
+
+	ch := make(chan (string))
+
+	// Print in the separate goroutine
+	go func() {
+		for line := range ch {
+			fmt.Println(line)
+		}
+	}()
+
+	// Printing is made this way to avoid time spent on generating line
```

Comment:
As we discussed an overhead of printing logs seems to be negligible. I don't have a strong opinion, though the code would be much simpler if you wouldn't care about the delay.